### PR TITLE
AIFC additional interface validation checks

### PR DESCRIPTION
**Pull request by D047539**
Additional validation checks are required when AIFC content is downloaded from abapGIT before anything is written to DB / persisted.
See Issue #5250

### CODE REVIEW

**Comment by larshp** [comment]
@D047539 check conflicts

**Comment by D047539** [comment]
> @D047539 check conflicts

remove AIF types

**Comment by larshp** [comment]
you should see this button,
image

File: src/objects/zcl_abapgit_object_aifc.clas.abap [hunk 1]
```diff
@@ -279,6 +286,10 @@ CLASS ZCL_ABAPGIT_OBJECT_AIFC IMPLEMENTATION.
     ENDIF.
 
     TRY.
+        IF execute_checks( io_xml = io_xml ) = abap_false.
```
Comment (mbtools) [inline src/objects/zcl_abapgit_object_aifc.clas.abap:289]: if there's just one parameter, please remove the parameter name: execute_checks( io_xml ). There're are few occurances like this. Thanks

**Review by mbtools — CHANGES_REQUESTED** [review]

**Comment by D047539** [comment]
Note 3117150 (https://launchpad.support.sap.com/#/notes/3117150) needs to be implemented (re-implemented)

**Comment by christianguenter2** [comment]
> Note 3117150 (https://launchpad.support.sap.com/#/notes/3117150) needs to be implemented (re-implemented)

Cannot implement note on 755 SP01. Whereas with previous version of the note it was possible.
grafik

**Comment by D047539** [comment]
> @D047539
> 
> > Note 3117150 (https://launchpad.support.sap.com/#/notes/3117150) needs to be implemented (re-implemented)
> 
> Cannot implement note on 755 SP01. Whereas with previous version of the note it was possible. grafik

I just released the note. It may take some time before the note is fully updated.

**Comment by christianguenter2** [comment]
@D047539 
I was able to implement the note. Is there a way to delete AIFC objects?

File: src/objects/zcl_abapgit_object_aifc.clas.abap [hunk 2]
```diff
@@ -520,4 +531,68 @@ CLASS ZCL_ABAPGIT_OBJECT_AIFC IMPLEMENTATION.
...
+                    iv_name = '/AIF/T_FINF'
+                  CHANGING
+                    cg_data = <lt_table> ).
+
+        READ TABLE <lt_table> ASSIGNING <ls_table> INDEX 1.
+        ASSIGN COMPONENT 'NS' OF STRUCTURE <ls_table> TO <lv_value>.
+        IF <lv_value> IS ASSIGNED.
+          ls_ifkeys-ns = <lv_value>.
+          UNASSIGN  <lv_value>.
+        ENDIF.
```
Comment (christianguenter2) [inline src/objects/zcl_abapgit_object_aifc.clas.abap:570]: Maybe simplify this and other places to
```suggestion
        IF sy-subrc = 0.
          ls_ifkeys-ns = <lv_value>.
        ENDIF.
```

File: src/objects/zcl_abapgit_object_aifc.clas.abap [hunk 3]
```diff
@@ -520,4 +531,68 @@ CLASS ZCL_ABAPGIT_OBJECT_AIFC IMPLEMENTATION.
...
+    DATA lr_structdescr TYPE REF TO cl_abap_structdescr.
+    DATA lr_table TYPE REF TO data.
+    FIELD-SYMBOLS <lt_table> TYPE STANDARD TABLE.
+    FIELD-SYMBOLS <ls_table> TYPE any.
+    FIELD-SYMBOLS: <lv_value> TYPE any.
+
+    DATA: lx_dyn_call_error TYPE REF TO cx_sy_dyn_call_error.
+    DATA: lx_root TYPE REF TO cx_root.
+
+    CLEAR lr_tabledescr.
```
Comment (christianguenter2) [inline src/objects/zcl_abapgit_object_aifc.clas.abap:549]: No need to clear local variable
```suggestion
```

File: src/objects/zcl_abapgit_object_aifc.clas.abap [hunk 4]
```diff
@@ -520,4 +531,68 @@ CLASS ZCL_ABAPGIT_OBJECT_AIFC IMPLEMENTATION.
...
+      zcx_abapgit_exception=>raise( 'Fieldsymbol not assigned' ).
+    ENDIF.
+
+    TRY.
+        io_xml->read( EXPORTING
+                    iv_name = '/AIF/T_FINF'
+                  CHANGING
+                    cg_data = <lt_table> ).
+
+        READ TABLE <lt_table> ASSIGNING <ls_table> INDEX 1.
```
Comment (christianguenter2) [inline src/objects/zcl_abapgit_object_aifc.clas.abap:565]: What if `READ` fails? If it never should fail I'd suggest to add `ASSERT sy-subrc = 0`

**Comment by D047539** [comment]
> @D047539 I was able to implement the note. Is there a way to delete AIFC objects?

Currently it is not possible to delete AIFC objects

**Comment by christianguenter2** [comment]
> > @D047539 I was able to implement the note. Is there a way to delete AIFC objects?
> 
> Currently it is not possible to delete AIFC objects

Hm, ok. Then I can't test.

**Comment by larshp** [comment]
well, if thats how AIFC works, @D047539 I guess its a limitation in the tooling? Is it described somewhere on help.sap.com?

**Comment by mbtools** [comment]
no delete = no unit tests, no ci tests. not sure if we want to introduce objects like this to abapGit. 

if there's no proper api, there's also batch input.

**Review by mbtools — APPROVED** [review]
The code looks ok but I have no way of testing it. Nicole, the missing delete feature is _not_ a show stopper but it's fairly limiting how useful abapGit will be for `AIFC`. If and when SAP adds deletion, please add it to abapGit as well via PR.

**Pull request merged**